core/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -142,6 +143,10 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
                         }
                         Store.tryOpenIndex(shardPath.resolveIndex(), shardId, nodeEnv::shardLock, logger);
                     } catch (Exception exception) {
+                        if (exception instanceof IOException && exception.getCause() instanceof ShardLockObtainFailedException) {
```

Comment:
question - why not let `Store.tryOpenIndex` throw `ShardLockObtainFailedException` and catch it here directly, logging it and not storing it as a "store" exception? The first part will make things simpler imo. I'm fine with not going with the second part and doing it like you propose but wanted to better understand your reasoning.